common/k8stest/util_snapshot.go

```diff
@@ -776,10 +775,30 @@ func GetMayastorSnapshotScMap() ([]string, error) {
 	snapScs, err := ListSnapshotClass()
 	if err == nil {
 		for _, vsc := range snapScs.Items {
-			if vsc.Driver == csiDriver {
+			if vsc.Driver == e2e_config.GetConfig().Product.CsiProvisioner {
 				mayastorSnapshotStorageClasses = append(mayastorSnapshotStorageClasses, vsc.Name)
 			}
 		}
 	}
 	return mayastorSnapshotStorageClasses, err
 }
+
+// CreateVolumeSnapshot create snapshot class and snapshot
+// it return snapshot object , snapshot content name and err
+func CreateVolumeSnapshot(snapshotClassName string, snapshotName string, pvc string, namespace string, csiDriver string) (*v1.VolumeSnapshot, string, error) {
+	// Get snapshot class before creating snapshot to verify snapshot class existence
+	//  and create if not present
+	vsc, err := GetSnapshotClass(snapshotClassName)
+	if (err != nil && k8serrors.IsNotFound(err)) || vsc == nil {
```

Comment:
It is more robust and flexible to fail if the snapshot class does not exist - instead of creating the snapshot class.

Reply:
The function is simpler as well, no need for csiDriver formal parameter

Reply:
Removed snapshot class check and always create snapshot class , This function needs csiDriver parameter because it will be different for different engines. https://github.com/openebs/openebs-e2e/compare/10b5596b7f253774426b97f1ad8a15fd26f40d9d..5b798e5d7c024c5b2018dafe3b623b04f08a6b00